crates/bevy_ecs/src/change_detection.rs

```diff
@@ -131,6 +131,11 @@ pub trait DetectChangesMut: DetectChanges {
     /// This is useful to ensure change detection is only triggered when the underlying value
     /// changes, instead of every time it is mutably accessed.
     ///
+    /// If you're dealing with non-trivial structs which have multiple fields of non-trivial size,
+    /// then consider applying a [`map_unchanged`](Mut::map_unchanged) beforehand to prevent the
+    /// need for unnecessary copying and cloning. See the docs of
+    /// [`map_unchanged`](Mut::map_unchanged) for an example
```

Comment:
I would specify that the idea is that after `map_unchanged` only the relevant field needs to be set, not the whole value. Also, concluded the last sentence with a dot.

```suggestion
    /// If you're dealing with non-trivial structs which have multiple fields of non-trivial size,
    /// then consider applying a [`map_unchanged`](Mut::map_unchanged) beforehand to allow
    /// changing only the relevant field and prevent unnecessary copying and cloning.
    /// See the docs of [`map_unchanged`](Mut::map_unchanged) for an example.
```